### ISSUE TITLE
TypeError: Object #<Object> has no method 'getTestBrowserInfo'

### ISSUE THREAD

**Issue by tchakabam** [browserstack/browserstack-runner#145]
more or less or often, we hit this issue ... 

doesn't seem to be related to our test but actually the runner just crashing at some point:

```

> browserstack-runner

Launching 1 worker(s) for 4 run(s).

<some app specific logs here ...>

/data/jenkins/slave/workspace/AudioManager_mse-fix-buffering-detection/node_modules/browserstack-runner/lib/server.js:239
Passed:') + ' %d tests, %d passed, %d failed; ran for %dms', worker.getTestBro
                                                                    ^
TypeError: Object #<Object> has no method 'getTestBrowserInfo'
    at reportHandler (/data/jenkins/slave/workspace/AudioManager_mse-fix-buffering-detection/node_modules/browserstack-runner/lib/server.js:239:140)
    at IncomingMessage.<anonymous> (/data/jenkins/slave/workspace/AudioManager_mse-fix-buffering-detection/node_modules/browserstack-runner/lib/server.js:305:46)
    at IncomingMessage.emit (events.js:92:17)
    at _stream_readable.js:943:16
    at process._tickCallback (node.js:419:13)

```

**Comment by shirish87** [comment]
Hey @tchakabam, I've been trying to trace this issue. Any clues on when this occurs?

**Comment by tchakabam** [comment]
I think it happens when there might be no more workers available ... ? But that's just a hunch.

It happens really often on one of the branches I am currently testing.

Maybe you can tell me a way to have the runner spit out some logs, and I could run my test suite with that enabled, and that could give you more clues? :)

**Comment by tchakabam** [comment]
Note: It might also happen with a very specific browser config. I'll try to reproduce and provide more details, in the meantime, waiting for your instructions to have the runner spit our some more debug info?

**Comment by shirish87** [comment]
`browserstack-runner --verbose` prints debug info.

**Comment by shirish87** [comment]
Could you please update to the latest version? Looks like this was fixed. Issue occurred in certain cases with Android emulators.

**Comment by tchakabam** [comment]
trying both your suggestions now, thanks :)

**Comment by shirish87** [comment]
@tchakabam, does the latest version work for you?

**Comment by tchakabam** [comment]
@shirish87 yes that fixed it, thanks :)

**Comment by tchakabam** [comment]
hey @shirish87 , unfortunately this happened again ... using version 0.4.1 of the runner. here is a full output a got:

```
# Invoking the BrowserStack runner
/data/jenkins/slave/workspace/audiomanager_master/node_modules/.bin/npm run browserstack-test | tee browserstack-test-log.txt

> @sc/audiomanager@3.3.1 browserstack-test /data/jenkins/slave/workspace/audiomanager_master
> browserstack-runner --verbose

Using config: /data/jenkins/slave/workspace/audiomanager_master/browserstack.json
Launching server on port: 8888
[Fri Jan 29 2016 15:23:07 GMT+0000 (UTC)] Launching tunnel
[Fri Jan 29 2016 15:23:09 GMT+0000 (UTC)] Tunnel launched
Launching 1 worker(s) for 4 run(s).
[Windows 8, Chrome 47.0, build/tests_static_basic.html] Launching
[Windows 8.1, Firefox 43.0] %c16:28:06.609%c (%c    +2ms%c) |%caudiomanager%c HTML5AudioPlayer (some_random_ID)state changed "idle", position: 0, duration: 0
[Windows 8.1, Firefox 43.0] %c16:28:06.609%c (%c    +0ms%c) |%caudiomanager%c HTML5AudioPlayer (some_random_ID)killing ...

/data/jenkins/slave/workspace/audiomanager_master/node_modules/browserstack-runner/lib/server.js:243
Passed:') + ' %d tests, %d passed, %d failed; ran for %dms', worker.getTestBro
                                                                    ^
TypeError: Object #<Object> has no method 'getTestBrowserInfo'
    at reportHandler (/data/jenkins/slave/workspace/audiomanager_master/node_modules/browserstack-runner/lib/server.js:243:140)
    at IncomingMessage.<anonymous> (/data/jenkins/slave/workspace/audiomanager_master/node_modules/browserstack-runner/lib/server.js:309:46)
    at IncomingMessage.emit (events.js:92:17)
    at _stream_readable.js:943:16
    at process._tickCallback (node.js:419:13)

npm ERR! Linux 3.16.0-4-amd64
npm ERR! argv "node" "/data/jenkins/slave/workspace/audiomanager_master/node_modules/.bin/npm" "run" "browserstack-test"
npm ERR! node v0.10.31
npm ERR! npm  v2.10.1
npm ERR! code ELIFECYCLE
npm ERR! @sc/audiomanager@3.3.1 browserstack-test: `browserstack-runner --verbose`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the @sc/audiomanager@3.3.1 browserstack-test script 'browserstack-runner --verbose'.
npm ERR! This is most likely a problem with the @sc/audiomanager package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     browserstack-runner --verbose
npm ERR! You can get their info via:
npm ERR!     npm owner ls @sc/audiomanager
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /data/jenkins/slave/workspace/audiomanager_master/npm-debug.log
# Extracting the screenshot URLs from the run log
cat browserstack-test-log.txt | grep -o "Screenshot: .*" | cut -f2- -d' ' > browserstack-test-screenshots.txt
wget -i browserstack-test-screenshots.txt -x -P screenshots
No URLs found in browserstack-test-screenshots.txt.
Warning: you have no plugins providing access control for builds, so falling back to legacy behavior of permitting any downstream builds to be triggered
```

**Comment by tchakabam** [comment]
this issue should probably be re-opened.

**Comment by vedharish** [comment]
Closing as fixed in #146
